### PR TITLE
Rewrite build system for mbed tests

### PR DESCRIPTION
**Pull request by andschwa**
The previous logic was clearly a copy of the logic used by the libc
tests (where a source file was `#include`d in a fake `main.c`) which was
unnecssarily complicated for these tests. It also defined things that
were irrelevant, didn't use `add_enclave`, and hard-coded several paths.

### CODE REVIEW

**Comment by andschwa** [comment]
This fixes #1016. Don't ask.

File: tests/mbed/CMakeLists.txt [hunk 1]
```diff
@@ -17,17 +17,16 @@ if (NOT "${build_type}" STREQUAL "DEBUG")
 	set(is_nondebug_build TRUE)
 endif()
 foreach(testcase ${alltests})
-	string(FIND "${testcase}" "selftest" found_selftest)
-	if(${found_selftest} GREATER 0 )
-		get_testcase_name(${testcase} name "../../3rdparty/mbedtls/mbedtls/programs/test/")
-		add_enclave_test(tests/mbed/${name} libmbedtest_host libmbedtest-${name}_enc)
+        string(COMPARE EQUAL "${testcase}" "selftest" found_selftest)
```
Comment (johnkord) [inline tests/mbed/CMakeLists.txt]: There's an extra space here at the start

Comment (andschwa) [reply tests/mbed/CMakeLists.txt]: Ah, the original file has tabs. I'm not bothering to change it for now.

**Pull request merged**